src/courses/mappers/04.md

```diff
@@ -0,0 +1,342 @@
+
+Here are some scenarios which demonstrate some key aspects of mappers:
+
+::: details Transferring IDs
+Say we have to transfer the credentials of an employee who is moving from Business A to Business B. The data formats the businesses use for IDs are as follows:
+```json
+// Business A
+{Name, DoB, Age, Title}
+
+// Business B
+{employee, employeeBirthday, employeeAge, jobTitle}
+```
+
+How do we transfer John's credentials from Business A to Business B?
+```json
+{Name: 'John Doe', DoB: 10-6-1992, Age: 32, Title: 'Security Technician'}
+```
+
+What we can do is create a mapping which correlates the items from Business A's ID scheme to Business B's:
+```json
+{employee: Name, employeeBirthday: DoB, employeeAge: Age, jobTitle: Title}
+```
+
+With this, we can then develop a mapper which takes John's credentials from Business A and transforms it to Business B's format as so:
+```json
+{Name: 'John Doe', DoB: 10-6-1992, Age: 32, Title: 'Security Technician'}
+
+ ||
+\  /
+ \/
+
+MAPPER
+
+ ||
+\  /
+ \/
+
+{employee: 'John Doe', employeeBirthday: 10-6-1992, employeeAge: 32, jobTitle: 'Security Technician'}
+``` 
+
+The important thing to note here is that mappers rely on underlying mappings which match semantically similar fields between two objects. These matches allow us to correctly convert each item in one object to the other.
+:::
+
+::: details Translating a foreign language
+Say we are given the job to translate the traditional Chinese sentence '我愛貓' into English.
+
+Without knowledge of the language, this is an impossible task.
+
+However, say we now also had a mapper which utilized the following mapping:
+```json
+{'愛': 'love', '我': 'I', '貓': 'cats'}
+```
+
+Now this task becomes much more easier. We can apply this mapper which will translate the sentence from '我愛貓' to 'I love cats' and vice versa.
+
+The important thing to note here is that mappers make translating between two objects trivial, which means that anyone can do it and we can automate mappers to perform object conversions for us.
+:::
+
+
+## What Is an OHDF Mapper?
+
+An OHDF mapper is a mapper that allows the underlying conversion infrastructure found in OHDF Converters to correlate certain objects or values from one overarching security tool data format to another overarching security tool data format. In the case of the SAF tool suite, these mappers allow for the conversion of any given data format to HDF (\*-to-HDF) and vice versa (HDF-to-\*) using the tools provided by the existing conversion infrastructure.
+
+OHDF mappers are built upon the schemas that define the involved data formats, and the mechanics of the mappers revolve around correlating each schema's fields as accurately as possible.
```

Comment:
i wish they were always built off of schemas but that's unfortunately not always the case.  we have to make it clear that you don't always need a schema to do a mapping.

i would make it something more generic like "OHDF mappers are built upon an understanding of the 3rd party format and the OHDF schema, and correlating the fields in each as accurately as possible."

as stated prior, we don't want to unnecessarily restrict ourselves with how we phrase things.  however, we also don't want to write text here that's so general or undefined that it becomes difficult for someone not knee-deep in the middle of things to understand what's going on.

Reply:
We mention this briefly in the section with DbProtect - @charleshu-8 maybe you can use some of the wording there

Reply:
Addressed. Also went ahead and attempted to scrub future instances of reliance on schema when not appropriate.